### PR TITLE
Add initial implementation of Update command

### LINKED ISSUE #92: Add update command
# Add update command
Format: `update COMPANY_INDEX APPLICATION_INDEX as/APPLICATION_STATUS`
due: 24 Oct 2024

### PR DESCRIPTION
**Pull request by blanklogic**
Resolves #92

### CODE REVIEW

File: src/test/java/seedu/internbuddy/logic/commands/UpdateCommandTest.java
Comment (Shux347) [inline src/test/java/seedu/internbuddy/logic/commands/UpdateCommandTest.java:1]: Tests are well-written and thorough. Thanks for keeping quality in mind.

File: src/main/java/seedu/internbuddy/logic/commands/WithdrawCommand.java [hunk 1]
```diff
@@ -36,6 +36,8 @@ public class WithdrawCommand extends Command {
      * @param applicationIndex of the application in the company to withdraw
      */
     public WithdrawCommand(Index companyIndex, Index applicationIndex) {
+        requireNonNull(companyIndex);
```
Comment (Shux347) [inline src/main/java/seedu/internbuddy/logic/commands/WithdrawCommand.java:39]: This is a solid improvement. Thanks for making it easier to maintain.

**Review by Shux347 — APPROVED** [review]

File: src/main/java/seedu/internbuddy/logic/commands/UpdateCommand.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+public class UpdateCommand extends Command {
+
+    public static final String COMMAND_WORD = "update";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Updates the status of an application. "
+            + "Parameters: COMPANY_INDEX (must be a positive integer) "
+            + "APPLICATION_INDEX (must be a positive integer) "
+            + PREFIX_APP_STATUS + "APP_STATUS\n"
+            + "Example: " + COMMAND_WORD + " 1 1 "
```
Comment (uniqly) [inline src/main/java/seedu/internbuddy/logic/commands/UpdateCommand.java:31]: Not sure if we want to make the indexes to have prefixes like in the `WithdrawCommand` in #108 to be consistent.

File: src/test/java/seedu/internbuddy/logic/commands/UpdateCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        AppStatus appStatus1 = new AppStatus("INTERVIEWED");
+        AppStatus appStatus2 = new AppStatus("OFFERED");
+        UpdateCommand updateCommand1 = new UpdateCommand(INDEX_FIRST_COMPANY,
+                Index.fromOneBased(1), appStatus1);
+        UpdateCommand updateCommand2 = new UpdateCommand(INDEX_SECOND_COMPANY,
+                Index.fromOneBased(1), appStatus2);
+
+        // different objects -> returns false
+        assertNotEquals(updateCommand1, updateCommand2);
+    }
```
Comment (uniqly) [inline src/test/java/seedu/internbuddy/logic/commands/UpdateCommandTest.java:98]: Good test cases, but I think the `equals_sameObject_returnsTrue` is not exactly testing equality for the same object but rather the another `UpdateCommandTest` instance with the same values; we should test for both. 

Would including a testing for different individual parameters be more concrete in `equals_differentObject_returnsFalse`? 

Perhaps unifying the `equals` test into one test with the 3 different cases could be better?

**Review by uniqly — CHANGES_REQUESTED** [review]
Overall solid work! But the testing could be more complete (not to say that it is not thorough). 😃

File: docs/UserGuide.md [hunk 4]
```diff
@@ -21,20 +21,23 @@ InternBuddy is a **desktop app for managing contacts, optimized for use via a Co
...
+| **Exit**     | `exit`                                                                                                                                                                            |
 | **Find**     | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find Apple Inc`                                                                                                                         |
-| **List**     | `list`                                                                                                                                                                            |
 | **Help**     | `help`                                                                                                                                                                            |
+| **List**     | `list`                                                                                                                                                                            |
+| **Update**   | `update c/COMPANY_INDEX app/APPLICATION_INDEX as/APPLICATION_STATUS`<br> e.g.,`update 1 1 as/OA`                                                                                  |
```
Comment (uniqly) [inline docs/UserGuide.md:32]: Minor typo. Should be `update c/1 app/1 as/OA` instead.

**Review by uniqly — APPROVED** [review]
LGTM! 👍

**Pull request merged**